theano/tensor/opt.py

```diff
@@ -5841,3 +5841,22 @@ def local_add_mul_fusion(node):
 def local_grad_clip(node):
     if isinstance(node.op, theano.gradient.GradClip):
         return node.inputs
+
+@register_canonicalize
+@register_stabilize
+@register_specialize
+@gof.local_optimizer([T.alloc])
```

Comment:
I think it is good to register it at the 3 places. But use [T.Alloc] instead of [T.alloc]